File: bin/dnsdata/org/bsdpanel/net/IpAddress.java
public class IpAddress extends Address {
	private int[] octet;
	private final String local = "127.0.0.1";
	
	public IpAddress() {
		 storeAddress(local);
	}
	
	public IpAddress(String ip) {
		ip.trim();
		
		try {
			ip = checkAddress(ip);
		} catch (IpAddressException e) {
			ip = local;
		}
		
		storeAddress(ip);
	}
	
	public IpAddress(IpAddress address) {
		octet = address.octet;
	}
	
	public String toString() {
		StringBuffer output = new StringBuffer();
		for (int i = 0; i < octet.length; i++) {
			output.append(octet[i]);
		}
		
		return output;
	}
    
    public static boolean isValid(final String address) {
        try {
            address = checkAddress(address);
            return true;
        } catch (IpAddressException e) {
            return false;
        }
    }
	
	private void storeAddress(final String address) {
		String parts[] = address.split(delimiter);
		
		for (int i = 0; i < parts.length; i++) {
			octet[i] = Integer.parseInt(parts[i]);
		}
	}
	
	protected String checkAddress(final String address) throws AddressException {
		String parts[] = address.split(delimiter);
		boolean valid = false;
		
		// Check to make sure the address has the right number of parts
		if (parts.length == 4) {
			// Check to make sure the octet parts are actually integers
			for (int i = 0; i < parts.length; i++) {
				if (isInt(parts[i])) {
					valid = true;
				} else {
					valid = false;
				}
			}
		}
		
		if (valid) {
			return address;
		} else {
			throw InvalidIpAddressException();
		}
	}
	
	private boolean isInt(final String integer) {
		int number;
		try {
			number = Integer.parseInt(integer);
			return true;
		} catch (Exception e) {
			return false;
		}
	}
}

File: EPR-GTG-dns-module/bin/dnsdata/src/class/net/IpAddress.cpp
#include <iostream>
#include <string>

#include "IpAddress.h"

bsdPanelNet::IpAddress::IpAddress() {
    octet[0] = 127;
    octet[1] = 0;
    octet[2] = 0;
    octet[3] = 1;
}

bsdPanelNet::IpAddress::IpAddress(std::string& addr) {
    // OK we need to split the address into its parts
    std::stack<std::string> parts = addr.split(deliniator);
    
    // Now we need to check to make sure the parts are valid
    for (int i = 0; i < parts.length(); i++) {
        try {
            unsigned int part = parts.index(i);
            if (part >= 0 && part <= 255) {
                // we have a valid number
                octet[i] = part;
            } else {
                // Throw an exception?
                throw bsdPanelNet::IpAddressException("Invalid Address Segment");
            }
        } catch () {
            delete [] octet;        // unsure if this is the correct way to clear the array
            throw bsdPanelNet::IpAddressException("Invalid Address Segment");
        }
    }
    
    // if they are valid lets stick them into the array
    // well we actually did that earlier
}